pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -251,10 +270,45 @@ func GatherContainerImages(i *Gatherer) func() ([]record.Record, []error) {
 			continueValue = pods.Continue
 		}
 
+		// Transform map into a list for sorting.
+		type tmpImageCount struct {
+			Image      string
+			StartMonth string
+			Count      int
+		}
+		imageCounts := []tmpImageCount{}
+		for img, countMap := range img2month2count {
+			for month, count := range countMap {
+				imageCounts = append(imageCounts, tmpImageCount{
+					Image:      img,
+					StartMonth: month,
+					Count:      count,
+				})
+			}
+		}
+
+		// Sort images from most common to least common.
+		sort.Slice(imageCounts, func(i, j int) bool {
+			return imageCounts[i].Count > imageCounts[j].Count
+		})
+
+		// Reconstruct the image information into the reported data structure.
+		contInfo := ContainerInfo{
+			Images:     ContainerImageSet{},
+			Containers: PodsWithAge{},
+		}
+		for i, img := range imageCounts {
```

Comment:
Because we would collect all image IDs and all Counts (sum of all counts per months) we would iterate imageCounts, but write individual records from each month in img2month2count for this image iD.

Reply:
If I change it as mentioned above, then sure, that would be the way to do it, but I feel like the code will become even less readable as a result. I will try to implement it and show you what it would look like.